src/mongocxx/test/change_streams.cpp

```diff
@@ -451,7 +449,7 @@ TEST_CASE("Documentation Examples", "[min36]") {
         // Start Changestream Example 3
         stdx::optional<bsoncxx::document::view> resume_token;
         change_stream stream = inventory.watch();
-        for (auto& event : stream) {
+        for (auto it = stream.begin(); it != stream.end(); it++) {
```

Comment:
I think the iterator should be avoided when possible. The _only_ reason I'm using it here is because `event` would be an unused variable otherwise. As a side note, I used the construct `(void)var;` because that test explicitly tested against a range-based loop so I had to keep it. It didn't seem as appropriate here but I'm open to suggestions.